From stern@rowland.harvard.edu Wed Oct 11 13:32:23 2006
Date: Wed, 11 Oct 2006 16:31:53 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>
Subject: usbcore: add call to cleanup routine
Message-ID: <Pine.LNX.4.44L0.0610111628530.6353-100000@iolanthe.rowland.org>

This patch (as801) adds a missing cleanup call to usbcore as well as
the missing function declarations.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/core/usb.c |    1 +
 drivers/usb/core/usb.h |    2 ++
 2 files changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/core/usb.c
+++ gregkh-2.6/drivers/usb/core/usb.c
@@ -1069,6 +1069,7 @@ static void __exit usb_exit(void)
 	usbdev_cleanup();
 	usb_hub_cleanup();
 	usb_host_cleanup();
+	usb_class_cleanup();
 	bus_unregister(&usb_bus_type);
 	ksuspend_usb_cleanup();
 }
--- gregkh-2.6.orig/drivers/usb/core/usb.h
+++ gregkh-2.6/drivers/usb/core/usb.h
@@ -28,6 +28,8 @@ extern int usb_major_init(void);
 extern void usb_major_cleanup(void);
 extern int usb_host_init(void);
 extern void usb_host_cleanup(void);
+extern int usb_class_init(void);
+extern void usb_class_cleanup(void);
 
 #ifdef	CONFIG_PM
 
